### ISSUE TITLE
Collision model

### ISSUE THREAD

**Issue by Try** [Try/OpenGothic#182]
Gothic 2 collision are pretty ticky - need to investigate.

#### Collision `npc`(human) vs `landscape`
1. Frontal
![изображение](https://user-images.githubusercontent.com/122894/127567049-d1f1a7c0-bf14-478c-ad1f-f4f56ccd60dd.png)
2. Side
![изображение](https://user-images.githubusercontent.com/122894/127566295-5f507794-2c3c-4113-a6a2-b871f5b3b9d5.png)
3. Rotate after side collision
![изображение](https://user-images.githubusercontent.com/122894/127566988-7220aeba-4aff-4a49-8892-f0ff1b2829bf.png)

#### Collision `npc`(snapper) vs `landscape`
1. Frontal
![изображение](https://user-images.githubusercontent.com/122894/127567154-b95698ea-7f76-4ba8-a508-3c139a2b49ce.png)
2. Side
![изображение](https://user-images.githubusercontent.com/122894/127567218-61167d2b-001b-4020-9020-855f61e2722c.png)

**Comment by Try** [comment]
From https://github.com/Try/OpenGothic/pull/575:

Game had some issues with npc-origin point and it interaction with collision. Probably it's time to get rid of `translateY` and related hacks and make OpenGothic more like vanilla on this aspect.

**Comment by Try** [comment]
Some testing of collisions:

<img width="591" height="707" alt="Image" src="https://github.com/user-attachments/assets/a0b21379-fbb9-4c92-86df-63e11e5eac09" /> 

<img width="591" height="707" alt="Image" src="https://github.com/user-attachments/assets/957a76ac-a0b4-4f35-93a4-1281cb1c6b4b" />

<img width="591" height="707" alt="Image" src="https://github.com/user-attachments/assets/c3c51da2-531d-4a41-befd-91e030674a4c" />

Running asymmetrical npc, wolf in that case, into a wall resulting at one type of collision.
Side-walk and then rotation - allows to dip into a wall.

After dip into a wall: no way to move forward, but side-move will push npc back from the wall.
Makes me think, that this can be not a 'real' collider proxy, but just a set of six rays to estimate distances.

**Comment by thokkat** [comment]
Tested Karibik mod again and lava is not killing the player (save1). From https://github.com/Try/OpenGothic/pull/575#issuecomment-1977418490 

> mod:
> https://www.worldofgothic.de/dl/download_404.htm
> I used 1.1 patch. Second link are voice lines (not needed).
> 
> save files:
> [save_slot_.zip](https://github.com/Try/OpenGothic/files/14486773/save_slot_.zip)

And while running around assert  `void MoveAlgo::assertStateChange(State): Assertion !dead failed.` triggered when killed during jump.

**Comment by Try** [comment]
Hi, @thokkat !

Thanks for pointing out!
I've changed `assert` - it was just wrong, as it mean to asset next state. For lava - need to have a look,probably trigger bounding boxes

**Comment by Try** [comment]
@thokkat 

I'm testing Karibik now: collision is triggered (for something at least), but doesn't reach `TouchDamage`. Do you know, if this issue been caused by https://github.com/Try/OpenGothic/pull/915 or it's been always broken?

**Comment by thokkat** [comment]
Issue was present before. Just brought this up because it wasn't mentioned explicitly here. Your reply back then for reference: https://github.com/Try/OpenGothic/pull/575#issuecomment-1987345571

Trigger is a triggerscript (two actually) which calls a damage function. Setting `p.y += -100` in `Npc::centerPosition()`  causes the trigger.

**Comment by Try** [comment]
It seem Karibik issue is similar to #911 - there is a script function, `EVT_KARIBIK_SECONDRING` that meant to be called on touch.

**Comment by thokkat** [comment]
Damage function is a different one and doesn't need `self` to be set.

```C
func void b_malokai_hoehle3_lava_func() {
    npc_changeattribute(hero, atr_hitpoints, -hero.attribute[atr_hitpoints]);
    pc_hero.effect = "SPELLFX_FIREARMOR";
};
```
Testing with increased trigger bbox height works and hero is killed, so should be "only" missing collision.

**Comment by Try** [comment]
I've changed bbox math to use "visual" bbox, instead of collision box. Basically: not sure how it meant to work, but seem more plausible, then arbitrary coordinate offset to Y

**Comment by thokkat** [comment]
> then arbitrary coordinate offset to Y

Yeah this was more meant as "proof" it's a collision only problem.

**Comment by Amanthuul** [comment]
Noticed that you can fall through both bridges at Khorinis entrances while walking normally.

Also can't swim under water, it pushes player back to the surface.

G2NoTR, v1.0.3846

**Comment by Try** [comment]
HI, @Amanthuul !

Khorinis and some bridges are indeed very difficult cases: they have no invisible polygon, for npc to walk on, so physics engine has to test 'land-ray' vs polygonal mesh, and this mesh has small crasks. 
In theory: many rays or linear-sweep-sphere may work, but at a extra performance cost..

> Also can't swim under water, it pushes player back to the surface.

Can't reproduce it, do you have save/screenshoot?

**Comment by Amanthuul** [comment]
https://github.com/user-attachments/assets/607fb6ba-afa6-49a0-96aa-6fa4f764b050

fitst time - holding alt, 2nd time - just pushed once, any body of water behaves like that

[save_slot_18.sav.txt](https://github.com/user-attachments/files/30719816/save_slot_18.sav.txt)

**Comment by Amanthuul** [comment]
Okay I'm just stupid, it's not collision problem, it's controls problem. Using mouse, I actually can swim under water, just control scheme is different from OG. 
Originally, up-down controls pitch, left-right control yaw when you under water, here up = forward and no pitch controls with keyboard.
